NationStatesAPIBot/NationStatesAPIBot/Services/DumpDataService.cs

```diff
@@ -1,6 +1,245 @@
-﻿namespace NationStatesAPIBot.Services
+﻿using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Linq;
+using DumpData;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NationStatesAPIBot.Types;
+
+namespace NationStatesAPIBot.Services
 {
     class DumpDataService
     {
+        private readonly BaseApiService _apiService;
+        private static List<NATION> _nations;
+        private static List<REGION> _regions;
+        private static Task _updating;
+
+        public DumpDataService(ILogger<RecruitmentService> logger, IOptions<AppSettings> appSettings, BaseApiService apiService, CancellationToken cancel)
+        {
+            _apiService = apiService;
+            PeriodicUpdate(TimeSpan.FromDays(1), cancel);
+        }
+        
+        private async Task<List<NATION>> GetNationsAsync()
+        {
+            await Task.WhenAll(_updating);
+            return _nations;
+        }
+        
+        private async Task<List<REGION>> GetRegionsAsync()
+        {
+            await Task.WhenAll(_updating);
+            return _regions;
+        }
+        
+        public async Task PeriodicUpdate(TimeSpan interval, CancellationToken cancel)
+        {
+            while (true)
+            {
+                await Update();
+                await Task.Delay(interval, cancel);
+            }
+        }
+
+        public Task Update()
+        {
+            return _updating = Task.Run(async () => {
+                var regionsStream = await _apiService.GetNationStatesDumpStream(NationStatesDumpType.Regions);
+                var nationsStream = await _apiService.GetNationStatesDumpStream(NationStatesDumpType.Nations);
+                _regions = GetRegionsFromCompressedStream(regionsStream);
+                _nations = GetNationsFromCompressedStream(nationsStream);
+                AddNationsToRegions();
+            });
+        }
+        
+        private static List<REGION> GetRegionsFromCompressedStream(GZipStream stream)
+        {
+            Console.WriteLine("Extracting compressed stream to REGION Collection");
```

Comment:
Please use logger instead of Console.WriteLine

Reply:
Totally overlooked those xD

Reply:
When logging please use LogEventId Provider and LogMessageBuilder see BasicNationStats for reference.